### PR TITLE
[Make] add -Werror, warnings as errors

### PR DESCRIPTION
**Pull request by larroy**

### CODE REVIEW

**Comment by lebeg** [comment]
I think there are a lot of warnings still to fix, until they can be treated as errors.

**Comment by KellenSunderland** [comment]
I think I'd prefer to turn on USE_DEV=1 for some of these builds in order to fail the clang builds when the detect some warnings.  I was planning on doing that and opening a PR that would address the warnings at the same time.

File: Makefile [hunk 1]
```diff
@@ -53,7 +53,7 @@ include mshadow/make/mshadow.mk
 include $(DMLC_CORE)/make/dmlc.mk
 
 # all tge possible warning tread
-WARNFLAGS= -Wall -Wsign-compare
+WARNFLAGS= -Wall -Wsign-compare -Werror
```
Comment (cjolivier01) [inline Makefile]: This is something that would need to be discussed on dev.  Also, when developing, you don't want to have errors on warnings

**Comment by larroy** [comment]
I wanted to see how bad it is, I'm fixing warnings all the time which sometimes cover bugs. I would like to have no warnings in one or two platforms at least.

**Comment by cjolivier01** [comment]
closing until it’s discussed in the community

**Pull request closed without merging**